modules/server_processes_manager/php/mriuploadserverprocess.class.inc

```diff
@@ -130,10 +130,25 @@ class MriUploadServerProcess extends AbstractServerProcess
      */
     public function getShellCommand()
     {
-        return "source {$this->_mriCodePath}/" . self::ENVIRONMENT_BASENAME . " ; "
-               . "{$this->_mriCodePath}/" . self::IMAGING_UPLOAD_FILE_PATH
-               . " -upload_id "    . escapeshellarg($this->_mriUploadId)
-               . " -profile prod " . escapeshellarg($this->_sourceLocation);
+        // Check that paths exist before using.
+        if (!is_dir($this->_mriCodePath)) {
+            throw new \ConfigurationException(
+                "Value for MRI code path is not a directory!"
+            );
+        }
+        if (!file_exists($this->_mriCodePath)) {
```

Comment:
Checking if the files exist after it as been check if it is a dir will never fail.
if it doesn't exist, it will fail previous test (is_dir).

either logic need to be change or ordering need to be changed